src/Fortran-interface/bml_getters_m.F90

```diff
@@ -41,7 +41,6 @@ subroutine bml_get_diagonal_single_real(a, diagonal)
     ptr = bml_get_diagonal_C(a%ptr)
     call c_f_pointer(ptr, diagonal_ptr, [bml_get_N(a)])
     diagonal = diagonal_ptr
-    deallocate(diagonal_ptr)
```

Comment:
We could possibly fix this by calling a C function to deallocate the pointer. Could you try that @mewall ?